e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js

```diff
@@ -504,106 +500,179 @@ describe("issues 20845, 25031", () => {
+      };
+
+      const dashboardDetails = {
+        name: "Dashboard with card with boolean field filter",
+        enable_embedding: true,
+        embedding_params: {
+          [dashboardParameter.slug]: "enabled",
         },
+        parameters: [dashboardParameter],
       };
-      createNativeQuestion(questionDetails)
-        .then(({ body: question }) => {
-          const DASHBOARD_FILTER_ID = "2850aeab";
-          const DASHBOARD_PARAMETER = createMockParameter({
-            id: DASHBOARD_FILTER_ID,
-            name: "Text filter for boolean field",
-            slug: "text_filter_for_boolean_field",
+
+      cy.get("@postgresInvoicesExpectedInvoiceId")
+        .then(fieldId => {
+          cy.createNativeQuestionAndDashboard({
+            questionDetails: getQuestionDetails(fieldId),
+            dashboardDetails,
           });
-          const dashboardDetails = {
-            name: "dashboard with card with boolean field filter",
-            enable_embedding: true,
-            embedding_params: {
-              [DASHBOARD_PARAMETER.slug]: "enabled",
-            },
-            dashcards: [
-              createMockDashboardCard({
-                id: -1,
-                size_x: 12,
-                size_y: 8,
-                card_id: question.id,
-                parameter_mappings: [
-                  {
-                    parameter_id: DASHBOARD_FILTER_ID,
-                    card_id: question.id,
-                    target: ["dimension", ["template-tag", TEMPLATE_TAG_NAME]],
-                  },
-                ],
-              }),
+        })
+        .then(({ body: dashboardCard }) => {
+          const { card_id, dashboard_id } = dashboardCard;
+
+          cy.wrap(dashboard_id).as("dashboardId");
+          cy.wrap(card_id).as("questionId");
+
+          const mapFilterToCard = {
+            parameter_mappings: [
+              {
+                parameter_id: dashboardParameter.id,
+                card_id,
+                target: ["dimension", ["template-tag", TEMPLATE_TAG_NAME]],
+              },
             ],
-            parameters: [DASHBOARD_PARAMETER],
           };
-          return createDashboardWithTabs(dashboardDetails);
-        })
-        .then(dashboard => {
-          cy.wrap(dashboard.id).as("dashboardId");
+
+          cy.editDashboardCard(dashboardCard, mapFilterToCard);
+
+          cy.request("POST", `/api/dashboard/${dashboard_id}/public_link`).then(
+            ({ body: { uuid } }) => {
+              cy.wrap(uuid).as("dashboardPublicUuid");
+            },
+          );
         });
     });
-  });
 
-  it("should allow dashboard filter to map to boolean field filter parameter in static embedding (metabase#27643)", () => {
-    cy.log("Test dashboard");
-    cy.get("@dashboardId").then(dashboardId => {
-      visitDashboard(dashboardId);
+    it("should allow a dashboard filter to map to a boolean field filter parameter in static embedding (metabase#27643)", () => {
```

Comment:
The title doesn't should reflect that we're testing both embedding and public dashboards in this test.